docs/assignment1.md

```diff
@@ -1,62 +1,58 @@
+The Setting class will include generic information about what objects, obstacles, etc. are in the game, as well as locations and directions from place to place. The location subclasses will hold the information for each specific room or location.
+The CommandParser class will handle the user’s commands, and determine whether an input is valid or not, given the object they are trying to use a command on. It will deal with any error handling that may be necessary.
+The Object class will contain all of the relevant information for different objects throughout the game, such as bikes, coffee, doors, etc. Each object will have a set of commands (from the Action class) that the user will be allowed to use on it.
+The TimeLimit class will keep track of how much time a user has left in the game, since the goal of our game will be to reach a certain location in a given period of time.
+The Obstacle class will contain information about certain obstacles the user will have to overcome to progress, and what actions they can use to get past it.
+Using these abstract classes and subclasses, we will import specific information to create a real game using JSON and a JSON parsing library, Fastjson. Players need to accomplish their objective within a certain time limit using commands, objects, and locations specified in a JSON file, which will be used to modify the abstract classes. We want to make it easy to use these abstract classes to make your own game, which will involve designing clear ways to input JSON files into a Main class.  If time permits, we want to implement a program which makes creating your own games extremely clear and easy, i.e. running a separate file in the terminal which asks which locations, objects, etc the user wants to use, and then creating a JSON and specific instance of a game based off of what the user wants.  We will also include a help command to specify what actions and objects can be used at a certain time to prevent confusion or getting lost.
+Other features we will implement if time permits are multiplayer and persistence. For multiplayer, we will take the game one step further by implementing a Java chat room using ServerSocket. Because text adventure games are basically semi-automated chat rooms between a player and the prewritten consequences of various actions and objects in the game, we believe we can add other users to this “chat room” to create multiplayer functionality which will depend on the overall objective of the game. The persistence feature will be significantly easier. In order to accomplish this, we will include a specification of where the player currently is and which objects they are holding in the JSON file that was initially inputted into the generic game. When the player closes the terminal, this information will save to this file, so if the same file is again inputted when the terminal is later opened, they can pick up where they left off.
 
```

Comment:
Clear goals and well articulated intro. Make sure to carefully adjust your scope when implementing extra features to avoid becoming too ambitious.